### PR TITLE
Remove use of ruby2_keywords

### PR DESCRIPTION
**Pull request by jonspalmer**
Remove the use of ruby2_keywords to create an `initialize` method that is better behaved with explicit keyword arguments

### CODE REVIEW

File: lib/dry/initializer/mixin/root.rb [hunk 1]
```diff
@@ -3,9 +3,13 @@ module Dry::Initializer::Mixin
   module Root
     private
 
-    def initialize(*args)
-      __dry_initializer_initialize__(*args)
+    def initialize(*args, **kwargs)
+      puts "root initialize: #{args}, #{kwargs}"
```
Comment (tmaier) [inline lib/dry/initializer/mixin/root.rb:7]: There is some debug code here

Comment (jonspalmer) [reply lib/dry/initializer/mixin/root.rb]: Fixed. Thanks for the catch

**Comment by jonspalmer** [comment]
Any chance we can approve the actions workflows here?

**Comment by jonspalmer** [comment]
I fixed the Rubocop linting issues. The jruby tests are failing but at a bundle install step which must be independent of this change.

**Comment by flash-gordon** [comment]
I don't follow. What does "better behave" mean?

**Comment by jonspalmer** [comment]
@flash-gordon the use of `ruby2_keywords` is something of a hack to help legacy code continue to work in the old way where the last positional arg could be used to pass keyword args. As of Ruby 3 keywords args must be passed explicitly. 

The specific problem is that before this fix if you introspect the constructor you get a misleading answer:

```ruby
require 'dry-initializer'

class User
  extend Dry::Initializer

  param  :name,  proc(&:to_s)
  param  :role,  default: proc { 'customer' }
  option :admin, default: proc { false }
  option :phone, optional: true
end

User.instance_method(:initialize).parameters #=> [[:rest, :args]]
```

After this change

```
User.instance_method(:initialize).parameters #=> [[:rest, :args], [:keyrest, :kwargs]]
```

I'll add a test that makes that clear. 

I also realize that for consistency we should have a better signature if the dry class doesn't accept positional args

**Comment by jonspalmer** [comment]
It would be great if the Initializer signature represented the state of the params. Ie if the dry-initialize doesn't include positional params the create an initialize method with just `**kwargs` similarly only `*args` for only positional args.

**Comment by jonspalmer** [comment]
Any chance we can get 👀 on this?

**Comment by jonspalmer** [comment]
A release would be awesome

**Comment by flash-gordon** [comment]
3.1.0 is out

**Pull request closed without merging**